lib/web_ui/dev/felt.dart

```diff
@@ -44,28 +46,13 @@ void main(List<String> args) async {
   } catch (e) {
     rethrow;
   } finally {
-    _cleanup();
+    await cleanup(browser: testCommand.browser);
```

Comment:
We should not run code in `TestCommand` when we run other commands. It breaks separation of concerns. For example, changing something in `TestCommand` can break other commands by accident, and the CI may not catch it because some commands are used for local testing only, such as `clean` and `build`, and some don't have anything to do with a "browser", such as `licenses`. It's also confusing. For example, the `browser` getter relies on `argResults`, but how does `TestCommand` get `argResults` if it never runs?

Suggestion: just like we have a global list of processes to clean up, we could also have a list of clean-up callbacks. A command can enqueue a callback to be called by the `cleanup` function. `TestCommand` enqueues a callback that cleans up Safari and only when Safari is used.

Reply:
Thanks this is a good suggestion.

Reply:
done. I added the aysnfunction you suggested offline.